File: apus.client/src/app/features/activities/ActivityDto/ActivityDto.ts
export interface ActivityDto {
    id: string;
    title: string;
    description?: string;
    duration: string;
    date: string;
    avgHr?: number;
    totalCalories?: number;
    type: string;
    displayname?: string;

    // gps fields 
    distanceKm?: number;
    elevationGain?: number;
    pace?: number;

    likescount?: number;
    isLikedByCurrentUser: boolean;

    userFullName: string;
    avatarUrl?: string;
}

export interface DisplayProp {
    label: string;
    value: string | number;
}

File: APUS.Server/Domain/DTOs/Feature/Activity/ActivityDto.cs
namespace APUS.Server.Domain.DTOs.Feature.Activity
{
	public record ActivityDto
	{
		public string Id { get; set; }
		public string Title { get; set; }
		public string? Description { get; set; }
		public TimeSpan Duration { get; set; }
		public DateTime Date {  get; set; }
		public int? AvgHr { get; set; }
		public int? TotalCalories { get; set; }
		public string Type { get; set; }
		public string? DisplayName { get; set; }
		public int? LikesCount { get; set; }
		public bool? IsLikedByCurrentUser { get; set; }

		//owner information
		public string UserFullName { get; set; }
		public string avatarUrl { get; set; }
	}

	public record GpsActivityDto : ActivityDto
	{
		public double? DistanceKm { get; set; }
		public double? ElevationGain { get; set; }
	}

	public record RunningActivityDto : GpsActivityDto
	{
		public double? Pace { get; set; }
	}
}
